src/environment.h

```diff
@@ -18,49 +18,60 @@
 
 #define MAX_ENV_VAR_LENGTH 2000
 
-static std::string get_env_variable(std::string env_variable)
+#include <sstream>
+
+static std::string get_env_variabe_no_log(
```

Comment:
Instead of taking an "out" parameter and returning a string value, would it make sense to return a std::pair (or std::tuple) with two string values - an error and an environment value? That way the calling signature is the same and you can change the error handling pattern to be `auto val = get_env_variable_no_log(xxx); if (!val.second.empty()) { log val.second; }`?